### ISSUE TITLE
Improvement of `plot_bounding_box_gallery` utility

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#1957]
**Short Description**

Demonstration with visual output is one of the crucial parts. Comparing the following two, I think pycoco-tools exhibit a bit more attractiveness. Maybe mostly for 

i. No black border around the box, 
ii. Box filled with color with lower opacity.

what if we improve from A to B?

| with keras_cv (A)    | with pycoco_tools (B) |
| -------- | ------- |
| <img src="https://github.com/keras-team/keras-cv/assets/17668390/3e366c38-eddd-41b1-8ecd-81c8afab54bb"> | <img src="https://github.com/keras-team/keras-cv/assets/17668390/56dc9d50-c166-405d-86c9-21b1416ed648" width=78%, > |

Some other features in the `plot_bounding_box_gallery` utility should have

- Notwithstanding the different classes, the box color annotation remains the same. It might become important for instance segmentation part.
- Return object (how to set title, x-label, y-label, etc)

**Existing Implementations**

- https://keras.io/guides/keras_cv/object_detection_keras_cv/

**Other Information**

The above data set is from: https://www.kaggle.com/datasets/ipythonx/global-wheat-challenge
The pycoco-tools is used: https://github.com/cocodataset/cocoapi/tree/master/PythonAPI/pycocotools

cc. @LukeWood

**Comment by LukeWood** [comment]
Tools like this should aim to be informative in my opinion; not necessarily pretty.  Personally I have a hard time finding all boxes on the right due to inconsistency and lack of high contrast lines.

**Comment by innat** [comment]
The above wheat example might be a weak example (single class with different color codes). 

The motivation came from the following image. The unique object has its own color code.
(Also, the box color fill looks super fancy, really not required, but not bad to have an optional argument, i.e. `box_fill`, etc)

<img src="https://github.com/keras-team/keras-cv/assets/17668390/25a82acc-6223-470c-b7dc-8ceaf9162546" width=70% />

Main objectives:

- Different color-code for unique objects. (make use of [class_mapping](https://github.com/keras-team/keras-cv/blob/3067132623cfa62b4b1f55ef30e3f37401fc99b4/keras_cv/visualization/plot_bounding_box_gallery.py#L43)). (another use cases: instance segmentation.)
- Return the plot object. 

---

Combining features from keras-cv and pycoco-tools.

```python
inputs = next(iter(tds.take(1)))
images, bounding_boxes = inputs["images"], inputs["bounding_boxes"]

image_nd = images.numpy().astype(np.uint8)
bbox_nd = bounding_boxes['boxes'].numpy()

get_plt = visualize_bbox(
    image_array=image_nd, 
    boxes_array=bbox_nd, 
    rows=1,
    cols=1,
    scale=10,
    from_type='coco', 
    box_random_color=False,
    box_fill=False
)
get_plt.show()
```

| example1    | example2  |
| -------- | ------- |
| <img src="https://github.com/keras-team/keras-cv/assets/17668390/fa9de8d6-9b0c-4cb9-931e-068b5a931c96" width=91%,> | <img src="https://github.com/keras-team/keras-cv/assets/17668390/f645d611-b94d-41d2-8f41-7d31e082c7c3"> |

**Comment by Zekrom-7780** [comment]
Hello @innat @LukeWood  , please could you assign this task to me so that I can try to improve it?

**Comment by innat** [comment]
@Zekrom-7780 

> returning the matplotlib object from our visualization functions. For example, if path is None and not show, we can return plt so that the user can do whatever they want with the plt object manually.

You can start working on it.

**Comment by Zekrom-7780** [comment]
@ianstenbit @innat 
Ok, I am SOO sorry for asking such Basic questions, but I need to ask them.
I have tried to return the plotted_images after the`plot_image_gallery` function is called, in the `plot_bounding_box_gallery.py`,
 as in 

```
  plot_image_gallery(
          plotted_images,
          value_range,
          legend_handles=legend_handles,
          rows=rows,
          cols=cols,
          **kwargs
      )
  return plotted_images
```

So How do I like test whatever I have committed in my branch of the forked repository? 
I did read `CONTRIBUTING.md` and set up the environment, but couldn't find anything regarding testing this.

Also is there like a slack channel, or mailing lists for asking doubts like this or this place itself is fine?

**Comment by ianstenbit** [comment]
> @Zekrom-7780
> 
> > returning the matplotlib object from our visualization functions. For example, if path is None and not show, we can return plt so that the user can do whatever they want with the plt object manually.
> 
> You can start working on it. cc. @ianstenbit

What exactly is the goal for this issue? To add color-coding to bounding boxes?

**Comment by ianstenbit** [comment]
> @ianstenbit @innat Ok, I am SOO sorry for asking such Basic questions, but I need to ask them. I have tried to return the plotted_images after the`plot_image_gallery` function is called, in the `plot_bounding_box_gallery.py`, as in
> 
> ```
>   plot_image_gallery(
>           plotted_images,
>           value_range,
>           legend_handles=legend_handles,
>           rows=rows,
>           cols=cols,
>           **kwargs
>       )
>   return plotted_images
> ```
> 
> So How do I like test whatever I have committed in my branch of the forked repository? I did read `CONTRIBUTING.md` and set up the environment, but couldn't find anything regarding testing this.
> 
> Also is there like a slack channel, or mailing lists for asking doubts like this or this place itself is fine?

I would test this by updating `examples/visualization/plot_image_gallery` to make use of the new feature and verify that it behaves as-expected.

**Comment by Zekrom-7780** [comment]
@ianstenbit 

> What exactly is the goal for this issue? To add color-coding to bounding boxes?

Yes, the goal of this issue is to add like different colours to each object that is detected, and to return the matplotlib object for the user, for any reason that they want it to use it.

Am I right?

**Comment by Zekrom-7780** [comment]
So i am able to return the matplotlib object `fig`  for `plot_bounding_box_gallery.py`. 

Now, for the color code part, @innat , you mentioned that 

> Different color-code for unique objects. (make use of [class_mapping](https://github.com/keras-team/keras-cv/blob/3067132623cfa62b4b1f55ef30e3f37401fc99b4/keras_cv/visualization/plot_bounding_box_gallery.py#L43)). (another use cases: instance segmentation.)

- So is the idea here that I need to create a new `Color` Dictionary, with random number of colors (and random values), equal to `len(class_mapping)`, and when each new class is detected, the color of the bounding box should be from the `Color` dictionary? 

- And this work should be in the `plot_bounding_box_gallery.py` ?

Am i thinking in the right direction? And if not, then how can I use `class_mapping`?

**Comment by Zekrom-7780** [comment]
@innat since Option B is done from my side, i'll send a PR for that 
And if option A  is needed (if it is decided that it is OK) , i'll work on that as well
